addons/website_sale/static/src/js/website_sale.js

```diff
@@ -638,6 +638,13 @@ export const WebsiteSale = publicWidget.Widget.extend(VariantMixin, cartHandlerM
         }
         return this._changeCountry();
     },
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onChangeState: function (ev) {
```

Comment:
You have to move the event too then

`        "change select[name='state_id']": "_onChangeState",`